internal/rest/avi_obj_string_group.go

```diff
@@ -160,11 +173,28 @@ func (rest *RestOperations) AviStringGroupCacheAdd(rest_op *utils.RestOp, vsKey
 			Uuid: uuid}
 
 		// StringGroup should not have a checksum
-		stringgroup_cache_obj.CloudConfigCksum = lib.StringGroupChecksum(keyvalues, stringgroup_cache_obj.Description, nil, false)
+		stringgroup_cache_obj.CloudConfigCksum = lib.StringGroupChecksum(keyvalues, nil, nil, nil, false)
 
 		k := avicache.NamespaceName{Namespace: rest_op.Tenant, Name: name}
 		rest.cache.StringGroupCache.AviCacheAdd(k, &stringgroup_cache_obj)
 
+		if strings.HasPrefix(name, lib.GetAKOUser()) {
+			vs_cache, ok := rest.cache.VsCacheMeta.AviCacheGet(vsKey)
+			if ok {
+				vs_cache_obj, found := vs_cache.(*avicache.AviVsCache)
+				if found {
+					vs_cache_obj.AddToStringGroupKeyCollection(k)
+					utils.AviLog.Debugf("Modified the VS cache for stringgroup object. The cache now is :%v", utils.Stringify(vs_cache_obj))
+				}
+
+			} else {
+				vs_cache_obj := rest.cache.VsCacheMeta.AviCacheAddVS(vsKey)
+				vs_cache_obj.AddToStringGroupKeyCollection(k)
+				utils.AviLog.Debug(spew.Sprintf("Added VS cache key during string group update %v val %v", vsKey,
```

Comment:
This can probably be rewritten as "Added VS cache key %v during string group update with val %v"

Reply:
Changed.